client/resource_manager_client.go

```diff
@@ -304,11 +304,13 @@ func (c *client) createTokenDispatcher() {
 		tokenBatchController: newTokenBatchController(
 			make(chan *tokenRequest, 1)),
 	}
+	c.wg.Add(1)
 	go c.handleResourceTokenDispatcher(dispatcherCtx, dispatcher.tokenBatchController)
 	c.tokenDispatcher = dispatcher
 }
 
 func (c *client) handleResourceTokenDispatcher(dispatcherCtx context.Context, tbc *tokenBatchController) {
+	defer c.wg.Done()
```

Comment:
Is it better to add a log at `case <-dispatcherCtx.Done():` to indicate the exit

Reply:
cc @CabinfeverB